apps/cowswap-frontend/src/modules/tradeQuote/state/tradeQuoteAtom.ts

```diff
@@ -25,26 +27,36 @@ export const DEFAULT_TRADE_QUOTE_STATE: TradeQuoteState = {
   localQuoteTimestamp: null,
 }
 
-export const tradeQuoteAtom = atom<TradeQuoteState>(DEFAULT_TRADE_QUOTE_STATE)
-
-export const updateTradeQuoteAtom = atom(null, (get, set, nextState: Partial<TradeQuoteState>) => {
-  set(tradeQuoteAtom, () => {
-    const prevState = get(tradeQuoteAtom)
-
-    // Don't update state if Fast quote finished after Optimal quote
-    if (
-      prevState.fetchStartTimestamp === nextState.fetchStartTimestamp &&
-      nextState.response &&
-      nextState.quoteParams?.priceQuality === PriceQuality.FAST
-    ) {
-      return { ...prevState }
-    }
-
-    return {
-      ...prevState,
-      ...nextState,
-      quoteParams: typeof nextState.quoteParams === 'undefined' ? prevState.quoteParams : nextState.quoteParams,
-      localQuoteTimestamp: nextState.response ? Math.ceil(Date.now() / 1000) : null,
-    }
-  })
-})
+export const tradeQuotesAtom = atom<Record<SellTokenAddress, TradeQuoteState | undefined>>({})
```

Comment:
Here we have the most important changes. Changes the state from single `TradeQuoteState` to a map.

Reply:
I didn't get why the key is the sell token?

Reply:
Basically, I replicated the state from the legacy: https://github.com/cowprotocol/cowswap/blob/release/2024-07-02/apps/cowswap-frontend/src/legacy/state/price/reducer.ts#L26

Sorry, I din't your question, what is supposed to be a key?
If you are worried about other parameters such as amount, kind, etc., the quote refetches when any of them are changed.
The caching only works when you switch token places or change sell token forth and back without amount changes (see "To Test" in the PR description.)